packages/block-library/src/image/index.php

```diff
@@ -69,27 +59,59 @@ function render_block_core_image( $attributes, $content, $block ) {
 	return $processor->get_updated_html();
 }
 
-
 /**
- * Add the directives and layout needed for the lightbox behavior.
+ * Add the lightboxEnabled flag to the block data.
  *
- * @param  string $block_content Rendered block content.
- * @param  array  $block         Block object.
- * @return string                Filtered block content.
+ * This is used to determine whether the lightbox should be rendered or not.
+ *
+ * @param  array $block Block data.
+ * @return array        Filtered block data.
  */
-function block_core_image_render_lightbox( $block_content, $block ) {
-	$link_destination = isset( $block['attrs']['linkDestination'] ) ? $block['attrs']['linkDestination'] : 'none';
+function block_core_image_should_render_lightbox( $block ) {
+
+	if ( 'core/image' !== $block['blockName'] ) {
+		return $block;
+	}
 
 	// Get the lightbox setting from the block attributes.
 	if ( isset( $block['attrs']['lightbox'] ) ) {
 		$lightbox_settings = $block['attrs']['lightbox'];
+		// If the lightbox setting is not set in the block attributes, get it from
+		// the global settings.
+	} else {
+		$lightbox_settings = gutenberg_get_global_settings( array( 'lightbox' ), array( 'block_name' => 'core/image' ) );
+
+		// If not present in global settings, check the top-level global settings.
+		if ( ! isset( $lightbox_settings['enabled'] ) ) {
+			$lightbox_settings = gutenberg_get_global_settings( array( 'lightbox' ) );
+		}
 	}
 
-	if ( ! isset( $lightbox_settings ) || 'none' !== $link_destination ) {
-		return $block_content;
+	$link_destination = isset( $block['attrs']['linkDestination'] ) ? $block['attrs']['linkDestination'] : 'none';
+
+	// If the lightbox is enabled and the image is not linked, flag the lightbox to be rendered.
+	if ( isset( $lightbox_settings['enabled'] ) &&
+	true === $lightbox_settings['enabled'] &&
+		'none' === $link_destination
+	) {
```

Comment:
Surprised the linter didn't catch this.

```suggestion
	if (
		isset( $lightbox_settings['enabled'] ) &&
		true === $lightbox_settings['enabled'] &&
		'none' === $link_destination
	) {
```